### ISSUE TITLE
Environment Variable Resolution is broken since Upgrade to IntelliJ 2016.3.2

### ISSUE THREAD

**Issue by markus-meisterernst** [esteinberg/plantuml4idea#139]
Hi,

in our team we recently chose PlantUML as our tool of choice. We resolve plantumls via Envrionment variable by defining a root variable SRC_HOME, so that every developer is free to choose the path to their respective project dir.

We have a global.puml that defines some Constants.
In a diagram X we import these Constants as such:

@startuml
!include %SRC_HOME%/docs/uml/global.puml
!include CMS_PACKAGE/diagrams/app-factory.puml
...

where CMS_PACKAGE is defined in global.puml refering to %SRC_HOME% as well.

In the IntelliJ Render View for diagram X I see an error with a path like such:

@startuml
"some path to where the puml file X is located"/%SRC_HOME%/docs/uml/global.puml
 Syntax Error?

I tried the plantuml.jar from the command line and it just works like a charme, so the problem seems to stem from the Upgrade from IntelliJ 2016.3.1 to 2016.3.2.
I'm using plantuml.jar version 2.4.0.

Could you, please, have a look at it.

Thank you very much.

Greets,

Markus

**Comment by krasa** [comment]
Have you tried command line on the one that is in ...\.IntelliJIdea16\config\plugins\plantuml4idea\lib
? Specifically plantuml.8051.jar  http://sourceforge.net/projects/plantuml/files/plantuml.8051.jar/download

**Comment by krasa** [comment]
It works for me 
![image](https://cloud.githubusercontent.com/assets/1160875/21761492/a12486c4-d653-11e6-8845-65bfdc7a8cbc.png)

Make sure to restart IJ after you set Envrionment variable.

**Comment by markus-meisterernst** [comment]
Hi Vojtěch, 

thx for your quick reply.
Based on your comment, I tried the following on the shell of my Mac (OS X, EL Capitan 10.11.6):

java -jar /Users/markus/Library/Application\ Support/IntelliJIdea2016.3/plantuml4idea/lib/plantuml.8051.jar compactsearch2.0.puml

and it worked. 

could be that this is a Mac related problem, as my colleagues have the same issue (all on Macs).
I tried to set the Environment variable through the login shell and the IDE itself as an Env Var, neither worked out.  By checking with my colleague I found out, that this problem seems to be IntelliJ version  agnostic (she is using IntelliJ 2016.2.5). 
I wounder why it worked before the IntelliJ update for me ....

My best guess would be that the problem is related to the path that has a space inside " " ( /Users/markus/Library/Application\ Support/IntelliJIdea2016.3/plantuml4idea/lib/plantuml.8051.jar), however, the global.puml renders correctly.

I attached you the screen shot of the message I get:

![screenshot 2017-01-09 11 54 47](https://cloud.githubusercontent.com/assets/2826134/21764386/116914f4-d663-11e6-9bc8-78a1bbe8d6ac.png)

It looks like PlantUML is using the path of the puml file and appends %EW_HOME%/docs/uml/global.puml to it.

Any Ideas ?

Markus

**Comment by krasa** [comment]
It looks like the env variable is not resolved correctly from IntellliJ VM.
You could try one thing, install https://plugins.jetbrains.com/idea/plugin/7282-liveplugin

and execute following script:
```
import static liveplugin.PluginUtil.*
show(System.getenv("EW_HOME"))
```
it would look like this, so you could check those variables of yours:
![image](https://cloud.githubusercontent.com/assets/1160875/21765180/86d21746-d667-11e6-9477-5ba2773f4ce0.png)

**Comment by markus-meisterernst** [comment]
Hi Vojtěch,

armed with your information, the liveplugin and the following Link
https://emmanuelbernard.com/blog/2012/05/09/setting-global-variables-intellij/

I was able to proof that the variable was actually not set. 

Furthermore I was able to actually set the variable using "launchctl" in a shell script to set the desired env variable EW_HOME. 

So, I can happly close the ticket.

Thanks a lot for your help, I really appreciate it.

Cheers,

Markus

Side-note: It's kind of annoying that the "Path Variables"-Settings under the IntelliJ Preferences doesn't work.
